meerk40t/tools/livinghinges.py

```diff
@@ -0,0 +1,1129 @@
+import wx
+
+from meerk40t.core.units import Length
+from meerk40t.gui.icons import icons8_hinges_50
+from meerk40t.gui.laserrender import LaserRender
+from meerk40t.gui.mwindow import MWindow
+from meerk40t.kernel import signal_listener
+from meerk40t.svgelements import (
+    Arc,
+    Close,
+    Color,
+    CubicBezier,
+    Line,
+    Matrix,
+    Move,
+    Path,
+    Point,
+    QuadraticBezier,
+)
+
+_ = wx.GetTranslation
+
+"""
+TODO:
+a) get rid of row / col range limitation and iterate until boundary exceeds frame
+b) Fix circle arc invocation
+c) Create proper bowlingpin shape
+d) Proper clipping
+e) Save presets per patterntype / or allow save / load of patternset
+f) Labels for additional parameters?!
+
+"""
+
+
+class LivingHinges:
+    """
+    This class generates a predefined pattern in a *rectangular* area
+    """
+
+    def __init__(self, xpos, ypos, width, height):
+        self.pattern = None
+        self.start_x = xpos
+        self.start_y = ypos
+        self.width = width
+        self.height = height
+        # We set it off somewhat...
+        self.gap = 0
+        self.x0 = width * self.gap
+        self.y0 = height * self.gap
+        self.x1 = width * (1 - self.gap)
+        self.y1 = height * (1 - self.gap)
+        self.param_a = 0.7
+        self.param_b = 0.7
+        # Requires recalculation
+        self.path = None
+        self.pattern = []
+        self.cutshape = ""
+        self.set_cell_values(10, 10)
+        self.set_padding_values(5, 5)
+        self.set_predefined_pattern("line")
+
+    def get_patterns(self):
```

Comment:
These should probably be dynamic, and store their name and the appended pattern in the same routine.

Reply:
Done